packages/go_router/lib/src/path_utils.dart

```diff
@@ -119,6 +119,28 @@ String concatenatePaths(String parentPath, String childPath) {
   return '${parentPath == '/' ? '' : parentPath}/$childPath';
 }
 
+/// Concatenates two Uri. It will [concatenatePaths] the parent's and the child's paths , then merges their query parameters.
+///
+/// e.g: pathA = /a?fid=f1, pathB = c/d?pid=p2,  concatenatePaths(pathA, pathB) = /a/c/d?fid=1&pid=2.
+Uri concatenateUris(Uri parentUri, Uri childUri) {
```

Comment:
the path looks good, but I am a bit hesitate on also merge query parameter. It has different logic from regular go. Regular go will not attempt to merge with current query parameters. I think we should probably follow the same logic to avoid confusion.

Reply:
Hmm this is quite interesting thing thou. Take this example.

We're in `/family/f2?sort=desc` and we're going to the inner `person/p1` route, and we'll pop it to go back to the `family` route. How are devs handling this case? 

In go_router_builder, devs can't use the easy `PersonRoute(pid: 'p1').go` but rather have to get its `route.location`, convert to Uri, and then manually add the sort param. Being able to use `Route().go` in builder is kinda a big selling point of the package imo

Reply:
you can access GoRouterState.of(context).uri.queryParemeters.

Also won't in this case PersonRoute also declare `sort` parameter?